### PR TITLE
[MRG] imshow transparency blend example

### PR DESCRIPTION
**Pull request by choldgraf**
Per a recent conversation on twitter, this is an example that shows off how you can blend transparency in with imshow when you're using a 2-D data matrix (as opposed to an image). A person was asking about how to do this on twitter, and as it's a common visualization in neuroscience I think it'd be helpful for people if they had an idea for how to get it done. I looked into supporting this with imshow directly, but it is not a trivial thing to do, so I figured documentation is the next best step. LMK what folks think.

### CODE REVIEW

File: examples/images_contours_and_fields/image_transparency_blend.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from matplotlib.colors import Normalize
+
+# Generate the space in which the blobs will live
+xmin, xmax, ymin, ymax = (0, 100, 0, 100)
+xx = np.linspace(xmin, xmax, 100)
+yy = np.linspace(ymin, ymax, 100)
+grid = np.array(np.meshgrid(xx, yy))
+grid = grid.transpose(2, 1, 0)
+
+# Generate the blobs
```
Comment (jklymak) [inline examples/images_contours_and_fields/image_transparency_blend.py]: Suggest specifying the expected data range of the raw blobs

File: examples/images_contours_and_fields/image_transparency_blend.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+################################################################################
+# Blending in transparency
+# ========================
+#
+# Below, we'll recreate the same plot, but this time we'll blend in
+# transparency with the image so that the extreme values are highlighted.
+# We'll also add in contour lines to highlight the image values.
+
+# Create an alpha channel based on weight values
```
Comment (jklymak) [inline examples/images_contours_and_fields/image_transparency_blend.py]: Suggest specifying that this means data with an absolute value below 0.00004 will get an alpha of 0.4 and that above 0.0010 will get an alpha value of 1.  I had to spend a couple of minutes figuring out what these two lines do.

File: examples/images_contours_and_fields/image_transparency_blend.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Blend transparency with color with 2-D images
+=============================================
+
+Blend transparency with color to highlight parts of data with imshow.
+
+A common use for :func:`matplotlib.pyplot.imshow` is to plot a 2-D statistical
+map. In this case, it's common to visualize the statistic of choice (e.g.,
+a t-statistic) alongisde another value of interest (e.g., the p-value for that
+statistic). One way to do this is to map the p-value onto the transparency of
+the image such that data with "significant" values are highlighted.
```
Comment (jklymak) [inline examples/images_contours_and_fields/image_transparency_blend.py]: I found this description a bit mysterious.  I'm not clear what is supposed to be "alongside" what.  I thought at first you were referring to the x and y axes, but I don't think you are.   

I'd simply say that its common to want to indicate significance of data values (or another quantity) using transparency overlain on an image.

**Comment by choldgraf** [comment]
took a stab at addressing comments...LMK what you think!

**Comment by choldgraf** [comment]
The 2.7 fail is mysterious to me... The error message isn't helpful in circle and it builds fine on my 2.7 environment :-/ good idea on the title! Will change when I'm at my computer

**Comment by choldgraf** [comment]
latest push has the new name in it

**Comment by choldgraf** [comment]
@QuLogic do you have any idea what this error could be on Circle 2.7? Not very informative and I can't recreate with 2.7 on my machine...

**Comment by dstansby** [comment]
Lets see if restarting the build does anything

**Comment by dstansby** [comment]
Maybe a sphinx-gallery issue, what version are you building with? CircleCI is sphinx-gallery=0.1.12

**Comment by QuLogic** [comment]
It looks like something in sphinx-gallery is trying to emit a warning, but doesn't pass the correct number of arguments, so it crashes. Unfortunately, sphinx hides everything but the final exception line, so it's very hard to say where the problem is right now.

**Comment by dopplershift** [comment]
It's a bug in sphinx 1.6.4, that was fixing a different messaging problem (submitted by @anntzer sphinx-doc/sphinx#4070). Forcing installation of sphinx 1.6.3 should fix it.

**Comment by choldgraf** [comment]
so that sounds like a change that should be made to the matplotlib sphinx build, not to this PR yeah? In that case nothing left to change here?

**Comment by dstansby** [comment]
I've added the change here because it's fairly small.

**Comment by choldgraf** [comment]
hmmm, 2.7 still failing. I'll try to debug with sphinx-gallery on my machine

**Comment by choldgraf** [comment]
I'm having a hard time getting the docs to build on my machine (just upgraded to a new computer and things are still wonky). Has anybody else been able to duplicate this sphinx bug? It doesn't seem related to this PR...

**Comment by QuLogic** [comment]
It looks like some kind of scipy/numpy inconsistency. Maybe something wrong on conda-forge.

**Comment by choldgraf** [comment]
so.....where are we on this one? 2.7 circle or appveyor aren't happy, though unclear if this is related to the PR. can we restart those builds?

**Comment by jklymak** [comment]
2.7 circleci is choking on `image_transparency_blend.py` so I think that is something you'll have to chase down in a 2.7 environment.  Not sure why....

**Comment by dstansby** [comment]
It looks to me like there's a problem with scipy.stats finding some numpy methods. Perhaps the easiest fix would be replacing calls to `multivariate_normal` by something like
```python
z = np.exp(-(x**2 + y**2))
z = z / np.sum(z)
```

**Comment by choldgraf** [comment]
latest push removes `multivariate_normal` (as well as any randomness so I didn't set a seed)

**Comment by choldgraf** [comment]
tests are passing!

**Comment by dstansby** [comment]
Should this be backported so it's on the website before 2.2 is released?

**Comment by tacaswell** [comment]
@meeseeksdev backport to v2.1.0-doc

**Comment by choldgraf** [comment]
@tacaswell this meeseecks thing is amazing

**Pull request merged**